tools/rvn/LogTrafficWatch.cs

```diff
@@ -0,0 +1,226 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Compression;
+using System.Net.WebSockets;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading;
+using System.Threading.Tasks;
+using Raven.Client.Documents.Changes;
+using Raven.Client.Extensions;
+using Raven.Client.Util;
+using rvn.Utils;
+using Sparrow;
+using Sparrow.Extensions;
+using Sparrow.Json;
+using Sparrow.Json.Parsing;
+using Sparrow.Utils;
+
+namespace rvn
+{
+    public class LogTrafficWatch : IDisposable
+    {
+        private readonly int _pid;
+        private ClientWebSocket _client;
+        private Timer _serverAliveCheck;
+        private int _exitCode;
+        private const int ServerAliveCheckInterval = 5000;
+        private CancellationTokenSource _cts = new CancellationTokenSource();
+
+        private readonly string _url;
+        private readonly X509Certificate2 _cert;
+        private readonly string _path;
+        private readonly List<TrafficWatchChangeType> _changeTypes;
+
+        public LogTrafficWatch(int? pid, string url, string certPath, string path, List<TrafficWatchChangeType>? changeTypes)
+        {
+            _pid = pid ?? ServerProcessUtil.GetRavenServerPid();
+            _path = path;
+            _changeTypes = changeTypes;
+            if (string.IsNullOrEmpty(certPath))
+            {
+                _url = url.StartsWith("http://") == false ? $"http://{url}" : url;
+            }
+            else
+            {
+                _cert = new X509Certificate2(certPath);
+                _url = url.StartsWith("https://") == false ? $"https://{url}" : url;
```

Comment:
Use `UriBuilder`, maybe? Or at least verify that we aren't going to get: `https://http://foo`